### PR TITLE
podvm-mkosi: set IMAGE_NAME instead of cp .git

### PR DESCRIPTION
**Pull request by mkulke**
Copying the .git folder can lead to problems, e.g. if the make process aborts, the extra .git folder is not removed and causes confusion in the working tree.

Since the IMAGE_NAME doesn't matter for the mkosi binary builds, we can set it explicitly to a static variable.

### CODE REVIEW

File: src/cloud-api-adaptor/podvm/Dockerfile.podvm_binaries.fedora [hunk 1]
```diff
@@ -36,6 +37,7 @@ ENV GUEST_COMPONENTS_REPO=${GUEST_COMPONENTS_REPO}
 ENV ATTESTER=${ATTESTER}
 ENV ARCH=${ARCH}
 ENV DEFAULT_AGENT_POLICY_FILE=${DEFAULT_AGENT_POLICY_FILE}
+ENV IMAGE_NAME=${IMAGE_NAME}
```
Comment (huoqifeng) [inline src/cloud-api-adaptor/podvm/Dockerfile.podvm_binaries.fedora:40]: If I understand correctly, are we going to drop the scripts in `/podvm/Makefile.inc`?
```
ifndef IMAGE_NAME
COMMIT := $(shell	commit=$$(git describe --match '' --dirty --always) && \
					if [ -n "$$(git ls-files --other --exclude-per-directory .gitignore "$(FILES_DIR)")" ]; then \
						commit="$${commit%-dirty}-dirty"; \
					fi && \
	                echo "$$commit")
ifndef COMMIT
$(error Failed to derive an image name. Explicitly define IMAGE_NAME)
endif
IMAGE_NAME := $(IMAGE_PREFIX)-$(COMMIT)-$(DEB_ARCH)
endif
```

Comment (mkulke) [reply src/cloud-api-adaptor/podvm/Dockerfile.podvm_binaries.fedora:40]: well, they will still be there, but we will go around them in an mkosi built, by setting IMAGE_NAME.

**Review by huoqifeng — APPROVED** [review]
LGTM

**Review by stevenhorsman — COMMENTED** [review]
So just to double check I understand this PR:
- We previously had the `.git` folder, so that we could generate the IMAGE_NAME in `Makefile.inc` via `git describe...`. We are now using the hard-coded `IMAGE_NAME` to skip the `git describe` logic instead. 
- IMAGE-NAME isn't needed in the podvm-mkosi flow, and is only user in the packer build? I guess this is where QiFeng's question came in that once packer is gone we'll just remove the IMAGE-NAME code altogether and can then remove this build-arg setting?

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks

**Pull request merged**